source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
-Writing your First Program
-==========================
+Writing your First Program:
+===========================
 
-This section is a detailed guide on how to build a `SIR Model <link for SIR URL>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
+This section is a detailed guide for a novice user on how to build a `SIR Model <https://bharatsim.readthedocs.io/en/latest/epidemiology-sir-model.html>`_ in BharatSim from scratch. By the end of this section, you should be able to write and execute a SIR model by yourself.
 
-Any model built using this framework contains different classes which are essentially extensions of different `Nodes <#>`_. To properly build a SIR model from scratch, you will first need to define these classes and the properties associated with them. These classes are grouped under different components of the model. Here, the SIR model has two different components:-
-1. `Agent <#>`_
-2. `Network <#>`_
+Any model built on this framework contains various classes which are essentially extensions of different Nodes. To build a SIR model from scratch, one needs to define the said classes and the properties and relationships associated with them. In the basic SIR there are two main components:
 
-These components are related to each other via simple relations which are described in detail later.
+1. `Agent`
+2. `Network`
 
-Setting up the components of the Network
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Agents are simply the individuals in the population and the Network is the manner in each of these individuals interact with one another. 
+However, to begin writing this program we assume that all the individuals are present in one location, hence neglecting the Network component. 

```

Comment:
No need of a "However" here. Just say something like:

"Here, we begin writing our program assuming that every agent is in contact with every other agent, which is equivalent to all of them being in the same "location".